composer/models/huggingface.py

```diff
@@ -114,3 +117,41 @@ def get_metrics(self, is_train: bool = False) -> Dict[str, Metric]:
 
     def update_metric(self, batch: Any, outputs: Any, metric: Metric) -> None:
         metric.update(outputs, self.labels)
+
+    def get_metadata(self):
+        model_output = {}
+        tokenizer_output = {}
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_dir = Path(tmp_dir)
+            model_dir = tmp_dir / 'model'
+            tokenizer_dir = tmp_dir / 'tokenizer'
+            self.model.config.save_pretrained(model_dir)
```

Comment:
Since we're using this API, is it worth doing some type checking in `__init__` to make sure the model and tokenizer match the input signature?

Reply:
I'm going to leave this as is. We don't typically explicitly raise errors for wrong types, and instead rely on type checking tools + it will just crash if the user uses the wrong type.